docs/guides.md

```diff
@@ -523,3 +523,307 @@ Now we enable autostart at the virtual machine level.
 5. Checking the output 
 `# xe vm-param-list uuid=<VM_UUID> | grep other-config`
 
+
+## Software RAID Storage Repository
+
+XCP-ng has support for creating a software RAID for the operating system but it is limited to RAID level 1 (mirrored drives) and by the size of the drives used.  It is strictly intended for hardware redundancy and doesn't provide any additional storage beyond what a single drive provides.
+
+These instructions describe how to add more storage to XCP-ng using software RAID and show measures that need to be taken to avoid problems that may happen when booting.
+
+These instructions assume you are starting with a server already installed with software RAID and have no other storage repositories defined except what may be on the existing RAID.
+
+### Example System
+
+The example system we're demonstrating here is a small server using 5 identical 1TB hard drives.  XCP-ng has already been installed in a software RAID configuration using 2 of the 5 drives.  There is already a default "Local storage" repository configured as part of the XCP-ng setup on the existing RAID 1 drive pair.
+
+Before starting the installation, all partitions were removed from the drives and the drives were overwritten with zeroes.
+
+So before starting out, here is an overview of the already configured system:
+
+```
+[09:51 XCP-ng ~]# cat /proc/partitions
+major minor  #blocks  name
+
+   8        0  976762584 sda
+   8       16  976762584 sdb
+   8       32  976762584 sdc
+   8       64  976762584 sde
+   8       48  976762584 sdd
+   9      127  976762432 md127
+ 259        0   18874368 md127p1
+ 259        1   18874368 md127p2
+ 259        2  933245487 md127p3
+ 259        3     524288 md127p4
+ 259        4    4194304 md127p5
+ 259        5    1048576 md127p6
+ 252        0  933232640 dm-0
+
+[09:51 XCP-ng ~]# cat /proc/mdstat
+Personalities : [raid1]
+md127 : active raid1 sdb[1] sda[0]
+      976762432 blocks super 1.0 [2/2] [UU]
+      bitmap: 1/8 pages [4KB], 65536KB chunk
+
+unused devices: <none>
+```
+
+The 5 drives are in place as sda through sde and as can be seen from the list are exactly the same size.  The RAID 1 drive pair is set up as the XCP-ng default of a partitioned RAID 1 array md127 using drives sda and sdb and is in a healthy state.
+
+### Building the Second RAID
+
+We have 3 remaining identical drives, sdc, sdd, and sde and we're going to create a RAID 5 array using them in order to maximize the amount of space.  We'll create this using the mdadm command like this:
+
+```
+[10:02 XCP-ng ~]# mdadm --create /dev/md0 --run --level=5 --bitmap=internal --assume-clean --raid-devices=3 --metadata=1.2 /dev/sdc /dev/sdd /dev/sde
+mdadm: array /dev/md0 started.
+```
+
+Here, we've made sure to use the assume-clean and metadata=1.2 options.  The assume-clean option prevents the RAID assembly process from initializing the content of the parity blocks on the drives which saves a lot of time when assembling the RAID for the first time.
```

Comment:
Writing zeroes is not a requirement for using the `assume-clean` option at all but is generally a good idea.  What the option does is to cause the RAID assembly to assume that there is no useful data on the drives and so not rebuild or, in this case, build parity information for the entire RAID array.  It's mainly a time saving measure to avoid having to wait for a RAID (re)build to finish before proceeding.